hnn_core/params.py

```diff
@@ -209,12 +209,17 @@ def _extract_drive_specs_from_hnn_params(params, cellname_list):
                                  key in cellname_list]
             sigma = par[cell_keys_present[0]][3]  # IID for all cells!
 
+            n_drive_cells = 'n_cells'
+            if par['sync_evinput']:
+                n_drive_cells = 1
+                drive['cell_specific'] = False
+
             drive['dynamics'] = {'mu': par['t0'],
                                  'sigma': sigma,
                                  'numspikes': par['numspikes'],
-                                 'sync_within_trial': par['sync_evinput']}
+                                 'n_drive_cells': n_drive_cells}
             drive['space_constant'] = par['lamtha']
-            drive['seedcore'] = par['prng_seedcore']
+            drive['seedcore'] = par['prng_seedcore'] - 18
```

Comment:
Is this the best way to modify the seedcore values to get the tests to pass?

Reply:
sure, but can you add a comment with the rationale? And maybe it needs an `if` statement with `legacy_mode` ?

Reply:
The "rationale" is weak, and many many small changes elsewhere might affect the precise value needed here. Do we _really_ need to mess with these?

Reply:
I'm happy to add a comment with rationale, but I don't think an `if` statement with `legacy_mode` is necessary since the function where this is called, `_extract_drives_specs_from_hnn_params()`, is by [definition a legacy artifact](https://github.com/jonescompneurolab/hnn-core/blob/40b07726a747c336debfe3a2683ee1607391a75d/hnn_core/params.py#L147-L150). It should also be noted that subtracting `18` from the seedcore values only provides consistency between hnn-gui and hnn-core when running `default.json`, not necessarily for other parameter configurations with a different number of bursty drives added.

@cjayb, if we wish to maintain consistency between hnn-gui and hnn-core (which is the whole point of legacy mode) for the default parameter set at least, we probably should. That said, if/when we decide to chuck legacy mode altogether, I'm more than happy to buy the first round :beers:

Reply:
sorry to comment on this further in what's already a stretched out PR. I see that this example is different: https://1828-168215891-gh.circle-artifacts.com/0/html/auto_examples/workflows/plot_simulate_alpha.html#sphx-glr-auto-examples-workflows-plot-simulate-alpha-py and my suspicion is that it's because of this hard-coded 18. Probably better to just adjust it in the examples as @cjayb had done before. 

I'm also all for removing `legacy_mode` ... just want it to be a separate clean PR so that there is a before/after. I think more fundamentally we have to 

1) Trust our unit tests more than the example outputs now. They test what is implemented rather than what we expect is the output
2) Allow variations due to seed. We talked about physiological plausibility in other contexts and in the same vein, the brain doesn't work by fixing the seed of the input spike trains. So, while they *can be* useful for debugging, I think changes in output due to differences seed have to be accepted in terms of educating the user about the sensitivity/stability of the model and the need to average multiple trials.

Reply:
Sounds good. Are you suggesting that we modify the seedcore value in the alpha example in this PR?

Reply:
yep, exactly. I'm guessing it's needed because `add_drives_from_params=False`

Reply:
Is the goal to make all of the examples look exactly the same as on master, or just certain ones?

Reply:
I don't think its possible to rectify the alpha example here with a change to the seedcore value because this PR changes the order that spike times get sampled. Previously, spike times were sampled for all `n_drive_cells` (previously `repeats`) on a given trial with a single call to

```python
t_array = prng.normal(np.repeat(isi_array, repeats), events_jitter_std)
```
For the alpha example, the random state was initialized once at `seedcore + trial_idx + gid = 14 + trial_idx + 270`.

Now, spike times are sampled individually for each drive cell where the random state is initialized based on the gid of each drive cell.

```python
t_array = prng.normal(isi_array, events_jitter_std)
```

Reply:
Since we're not referencing specific figures in the hnn-gui tutorials, is it good enough to just go through the tutorials and make sure the seedcore values are set so that whatever random behavior emerges supports the message of the tutorial?

Reply:
cool, understood. So I guess there is really no way to reproduce GUI outputs now with rhythmic inputs because of this change. Let's merge and move on. I agree this is already a great improvement!